diff --git a/node_modules/did-jwt/lib/index.cjs b/node_modules/did-jwt/lib/index.cjs
index 85c93dd..d79e0fd 100644
--- a/node_modules/did-jwt/lib/index.cjs
+++ b/node_modules/did-jwt/lib/index.cjs
@@ -1701,8 +1701,11 @@ const verifyJWT = function (jwt, options = {
       }
 
       didUrl = payload.client_id;
-    } else {
-      didUrl = payload.iss;
+    } else if (payload.iss?.indexOf('did:') === 0) {
+      didUrl = payload.iss
+    } else if (header.kid?.indexOf('did:') === 0) {
+      // OID4VCI expects iss to be the client and kid, to be the DID VM
+      didUrl = (header.kid || '').split('#')[0]
     }
 
     if (!didUrl) {
diff --git a/node_modules/did-jwt/lib/index.module.js b/node_modules/did-jwt/lib/index.module.js
index e5acf21..0b57279 100644
--- a/node_modules/did-jwt/lib/index.module.js
+++ b/node_modules/did-jwt/lib/index.module.js
@@ -1697,8 +1697,11 @@ const verifyJWT = function (jwt, options = {
       }
 
       didUrl = payload.client_id;
-    } else {
-      didUrl = payload.iss;
+    } else if (payload.iss?.indexOf('did:') === 0) {
+      didUrl = payload.iss
+    } else if (header.kid?.indexOf('did:') === 0) {
+      // OID4VCI expects iss to be the client and kid, to be the DID VM
+      didUrl = (header.kid || '').split('#')[0]
     }
 
     if (!didUrl) {
diff --git a/node_modules/did-jwt/lib/index.umd.js b/node_modules/did-jwt/lib/index.umd.js
index 2113790..41c0dc7 100644
--- a/node_modules/did-jwt/lib/index.umd.js
+++ b/node_modules/did-jwt/lib/index.umd.js
@@ -1691,8 +1691,11 @@
         }
 
         didUrl = payload.client_id;
-      } else {
-        didUrl = payload.iss;
+      } else if (payload.iss?.indexOf('did:') === 0) {
+        didUrl = payload.iss
+      } else if (header.kid?.indexOf('did:') === 0) {
+        // OID4VCI expects iss to be the client and kid, to be the DID VM
+        didUrl = (header.kid || '').split('#')[0]
       }
 
       if (!didUrl) {
diff --git a/node_modules/did-jwt/src/JWT.ts b/node_modules/did-jwt/src/JWT.ts
index 2978979..b2a1049 100644
--- a/node_modules/did-jwt/src/JWT.ts
+++ b/node_modules/did-jwt/src/JWT.ts
@@ -533,8 +533,11 @@ export async function verifyJWT(
       throw new Error(`${JWT_ERROR.INVALID_JWT}: JWT client_id is required`)
     }
     didUrl = payload.client_id
-  } else {
+  } else if (payload.iss?.indexOf('did:') === 0) {
     didUrl = payload.iss
+  } else if (header.kid?.indexOf('did:') === 0) {
+    // OID4VCI expects iss to be the client and kid, to be the DID VM
+    didUrl = (header.kid || '').split('#')[0]
   }
 
   if (!didUrl) {
